tests/lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,1246 @@
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b\n} fn();",
+            output: "a &&= b; fn();",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Spacing
+        {
+            code: "if (a) a  =  b",
+            output: "a  &&=  b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a)\n a = b",
+            output: "a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) {\n a = b; \n}",
+            output: "a &&= b;",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Comments
+        {
+            code: "/* before */ if (a) a = b",
+            output: "/* before */ a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) a = b /* after */",
+            output: "a &&= b /* after */",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) /* between */ a = b",
+            output: null,
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) a = /* between */ b",
+            output: null,
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Members
+        {
+            code: "if (a.b) a.b = b",
+            output: "a.b &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a[b].c) a[b].c = d",
+            output: "a[b].c &&= d",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "with (object) if (a.b) a.b = b",
+            output: "with (object) a.b &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Else if
+        {
+            code: "if (unrelated) {} else if (a) a = b;",
+            output: "if (unrelated) {} a &&= b;",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
```

Comment:
This fix seems wrong without `else`. In the original code, when `unrelated` is falsy there will be no assignment regardless of `a`, but in the fixed code `unrelated` does not affect whether or not `a &&= b;` will be executed.